### PR TITLE
Blogger - Build Schema structure

### PR DESCRIPTION
**Pull request by enocol**
In this mile stone, the following was achieved

A new rails project was setup call Blog_post using rails
Postgresql data was setup for the project
Model, Views and Controllers for Users, Posts, Comments and Likes Classes were created
Indexs were added to each field that is a foriegn key
Migrations were done for Users, Posts, Comments and Likes to Postgresql

### CODE REVIEW

File: db/migrate/20240110235639_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateUsers < ActiveRecord::Migration[7.1]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :Photo
+      t.string :Bio
+      t.integer :PostCounter
```
Comment (zeff96) [inline db/migrate/20240110235639_create_users.rb]: - [x] Hello, fantastic progress! 🌟 When working with Rails, it's customary to use snake_case for column names. In the given migration, I recommend changing the column names for "Photo", "Bio" and "PostCounter" to follow this convention:

```suggestion
      t.string :photo
      t.string :bio
      t.integer :posts_counter
```

File: db/migrate/20240111000127_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CreatePosts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.string :text
+      t.string :CommentsCounter
+      t.string :LikesCounter
```
Comment (zeff96) [inline db/migrate/20240111000127_create_posts.rb]: - [x] Hello, fantastic progress! 🌟 When working with Rails, it's customary to use snake_case for column names. In the given migration, I recommend changing the column names for "CommentsCounter" and "LikesCounter" to follow this convention:

```suggestion
      t.string :comments_counter
      t.string :likes_counter
```

File: db/migrate/20240111000127_create_posts.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class CreatePosts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.string :text
```
Comment (zeff96) [inline db/migrate/20240111000127_create_posts.rb]: - [x] Hello, awesome work so far! 🌟 To enhance user flexibility in inputting longer text without running into database limitations, consider changing the data type for the "text" column from string to the more accommodating text type in your Rails 

```suggestion
      t.text :text
```

File: db/migrate/20240111000939_create_comments.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class CreateComments < ActiveRecord::Migration[7.1]
+  def change
+    create_table :comments do |t|
+      t.string :text
+      t.bigint :UserId
+      t.bigint :PostId
```
Comment (zeff96) [inline db/migrate/20240111000939_create_comments.rb]: - [x] Hello, you're doing great! 🌟 In alignment with Rails conventions, let's ensure we use snake_case naming for the table columns. Specifically, for consistency, modify the column names in your migration to adhere to this convention:

File: db/migrate/20240111001706_create_likes.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CreateLikes < ActiveRecord::Migration[7.1]
+  def change
+    create_table :likes do |t|
+      t.bigint :UserId
+      t.bigint :PostId
```
Comment (zeff96) [inline db/migrate/20240111001706_create_likes.rb]: - [x] Hello, you're doing great! 🌟 In alignment with Rails conventions, let's ensure we use snake_case naming for the table columns. Specifically, for consistency, modify the column names in your migration to adhere to this convention:

File: db/migrate/20240111000939_create_comments.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class CreateComments < ActiveRecord::Migration[7.1]
+  def change
+    create_table :comments do |t|
+      t.string :text
```
Comment (zeff96) [inline db/migrate/20240111000939_create_comments.rb]: - [x] Hello, awesome work so far! 🌟 To enhance user flexibility in inputting longer text without running into database limitations, consider changing the data type for the "text" column from string to the more accommodating text type in your Rails

```suggestion
      t.text :text
```

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @enocol , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow
:heavy_check_mark: All migrations added

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20240111062838_create_users.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+      t.text :text
+      t.integer :comments_counter
+      t.string :likes_counter
+      t.bigint :author_id
+
+      t.timestamps
+      t.index :author_id
+      t.foreign_key :users, column: :author_id, primary_key: :id, on_delete: :cascade
+    end
+  end
```
Comment (danmainah) [inline db/migrate/20240111062838_create_users.rb]: - [ ] @enocol Great work implementing the changes requested, however you have used the wrong column names for the users table kindly consider changing the columns and using the correct columns.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @enocol , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Implemented the changes requested to use the recommended snake_case for column names.
✔️  Changed the data type to text in text column as requested to ensure the app can handle big text inputs.


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @enocol , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to ensure the users' table has the correct columns.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**